r-package/grf/bindings/RcppUtilities.h

```diff
@@ -14,10 +14,6 @@ class RcppUtilities {
    * Converts the provided {@link Forest} object and OOB predictions to an R list
    * to be returned through the Rcpp bindings. The provided predictions vector can
    * be present if OOB predictions were not requested as part of training.
-   *
-   * NOTE: To conserve memory, this method destructively modifies the forest
```

Comment:
My memory is not totally clear, but I think this note still applies since `RcppUtilities::serialize_forest` can destructively modify the forest?

Reply:
Ah, yes I was taking this note too literally, there is no "explicit" clearing being done,

https://github.com/grf-labs/grf/blob/master/r-package/grf/bindings/RcppUtilities.cpp#L55

BUT it loops over by fetching a unique_ptr to the invividual trees, since the pointer is unique the object dies after it goes out of scope and the original forest is empty.

Yes, that makes sense, that note is technically correct - I was just confused by the semantic.

I can add it back and note that it _implicitly_ clears out the forest since it moves all the unique tree pointers - or do you think that is too obvious?

Reply:
(can restore the original docstring comment and just a note on this in the code moving the tree ptr?)